xla/service/gpu/gpu_executable.cc

```diff
@@ -202,12 +203,25 @@ Status ExecuteThunks(const std::string& module_name, ModuleIdentifier module_id,
                      const ThunkSequence& thunk_sequence,
                      const ServiceExecutableRunOptions* run_options,
                      const BufferAllocations& buffer_allocations,
-                     bool block_host_until_done) {
+                     bool block_host_until_done,
+                     const DebugOptions& debug_options) {
   se::Stream* main_stream = run_options->stream();
   se::StreamExecutor* executor = main_stream->parent();
-
+  int high_priority = 0;
+  if (debug_options.xla_gpu_enable_highest_priority_async_stream()) {
+    CUresult res = cuCtxGetStreamPriorityRange(nullptr, &high_priority);
```

Comment:
This CUDA specific code seems duplicated below and will not work on other targets like AMD potentially. Can this be guarded by GOOGLE_CUDA?

Reply:
guarded with a directive and removed the cuda specific creation logic from here.